src/inlay-hints.ts

```diff
@@ -113,70 +116,47 @@ class InlayHintsFeature implements vscodelc.StaticFeature {
     }
   }
 
-  onDidChangeVisibleTextEditors() {
+  onDidCloseTextDocument(document: vscode.TextDocument) {
     if (!this.enabled)
       return;
+    
+    // Drop state for any file that is now closed.
+    const uri = document.uri.toString();
+    const file = this.sourceFiles.get(uri);
+    if (file) {
+      file.inlaysRequest?.cancel()
+      this.sourceFiles.delete(uri);
+    }        
+  }
 
-    const newSourceFiles = new Map<string, FileEntry>();
+  onDidChangeVisibleTextEditors() {
+    if (!this.enabled)
+      return;
 
-    // Rerender all, even up-to-date editors for simplicity
+    // When an editor is made visible we have no inlay hints.
```

Comment:
> Do the rendered decorations go away when an editor is no longer visible

Sadly yes, that's what I observed.

> This patch seems to make them appear quicker (or maybe it's just my imagination)

Previously they had to be fetched *from the server*, now they just get reapplied from the local cache.

> it's still a noticeable flash rather than them being there already
I guess the vscode inlay hints API is the way to fix this.

Reply:
Makes sense, thanks